app/Models/Cart.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Cart extends Model
+{
+    protected $guarded = [];
+
+    public static function addItem($userId, $product, $quantity = 1)
+    {
+        //if quantity is less than 1, update quantity to 0
+        $quantity = $quantity > 0 ? $quantity : 0;
+
+        // if item exist, just add quantity
+        $cart = self::exist($userId, $product['product_id'])->first();
+        if ($cart) {
+            $cart->increment('quantity', $quantity);
+            return $cart;
+        }
+
+        // do nothing if $quantity is less than 1
+        if ($quantity < 1) {
+            return null;
+        }
```

Comment:
Is this acceptable?
`$quantity < 1 ? null : $quantity;`

Reply:
nope in this case we should do nothing if $quantity is less than 1